Entry for challenge using unity

DerekGooding opened the pull request:
My solution is pretty simple. 

You only need a script on the player. You make a layer for things that can be dropped through. I called it "ground" in my example. The player script has a serializable field for the layer you choose so just added it to the prefab and set your platform prefab to use that layer. 

When a player initiates drop through, raycast and returns all colliders that have the stipulated layer. GetComponent the platforms collider, and set the collider to .trigger = true for a short time with a coroutine, then return it to false. 

It's about 20 lines of meaningful code. The rest is just general movement and respawning if you fall out of bounce and changing the platforms color when drop through for visualization. 

You don't need a script on the platforms using this solution. Just set them to the correct layer of your choosing.

DerekGooding committed:
Create Unity.gitignore

DerekGooding committed:
Move gitignore

DerekGooding committed:
Create .gitignore

DerekGooding committed:
Delete Unity.gitignore

DerekGooding committed:
Delete .gitignore

DerekGooding committed:
Create Unity.gitignore

DerekGooding committed:
Fix ignore

DerekGooding committed:
Init

DerekGooding committed:
Add jump back up

Also added color for if platform is fallthrough or not.

DerekGooding committed:
Added 2dphysics material

Removes the sticky walls issue

DerekGooding commented:
I fixed the detection issue with a 2d material.

MooNiZZ commented:
Great submission!
I would suggest updating the Drop method to use Physics2d.IgnoreCollision() instead of setting the collider to isTrigger to cover more use-cases.
e.g. 
```
    private IEnumerator Drop(Collider2D collider)
    {
        Physics2D.IgnoreCollision(collider, GetComponent<Collider2D>(), true);
        yield return new WaitForSeconds(0.5f);
        Physics2D.IgnoreCollision(collider, GetComponent<Collider2D>(), false);
    }
```
And preferably cache the GetComponent as well so we don't have to do that every time!
Keep up the good work!

DerekGooding committed:
Update PlayerController.cs

DerekGooding committed:
Update PlayerController.cs

Handle platform color change in the PlayerController. 

Can Remove ColorIfCollidable script entirely.

DerekGooding committed:
Update ColorIfCollidable.cs

Redundant
